### DISCUSSION TITLE
Transphone usage?

### DISCUSSION THREAD

**Discussion by SyGen899** [coqui-ai/TTS#2309] [General Q&A] [answered]
Hi, hoping someone can help out, I found this model https://github.com/xinjli/transphone for creating phonemes, and it covers a wider range of languages as opposed to espeak or gruut, I wanted to know if anyone has some pointers on how I can use this with the TTS package, can I generate those outputs and load at inference somehow or do I have to rewrite the phoneme wrappers to integrate this?

**Comment by lexkoro** [comment, marked as answer]
You would have to add it as a phonemizer. You can check some phonemizer implementations as reference here https://github.com/coqui-ai/TTS/tree/dev/TTS/tts/utils/text/phonemizers

**Comment by erogol** [comment]
Interesting library. Would be a nice PR.

**Comment by francois-vz** [comment]
Hi @SyGen899. It is also quite simple to generate the outputs and hack the source code to read them in. E.g. in ./TTS/TTS/tts/utils/text/phonemizers/base.py

```python
    def phonemize(self, text: str, separator="|", language: str = None, fn: str = None) -> str:  # pylint: disable=unused-argument
        phonemized = []
        prons_as_list = []
        prons = []
        df = pd.read_csv(
            "file.txt", 
            sep = '|',
            names = ['fn', 'phones'],
            ) # fn|string of IPA phones
        if fn is None:
            return "" # Insert eval sentences here
        list_ = df[df.fn==fn].phones.item().split()
        text, punctuations = self._phonemize_preprocess(text)
        idx=0
        for text_word in text:
            for _ in text_word.split(' '):
                prons.append(list_[idx])
                idx+=1
            if len(prons) > 1:
                string=' '.join(prons)
            else:
                string = prons[0]
            prons=[]
            prons_as_list.append(string)
        phonemized = self._phonemize_postprocess(prons_as_list, punctuations)
        return phonemized
```

**Reply by francois-vz** [reply]
You probably only want to do this way if you're in a rush and just want to test something.
